src/World.cpp

```diff
@@ -1130,19 +1130,46 @@ void cWorld::TickMobs(std::chrono::milliseconds a_Dt)
 		}  // for i - AllFamilies[]
 	}  // if (Spawning enabled)
 
-	// move close mobs
-	cMobProximityCounter::sIterablePair allCloseEnoughToMoveMobs = MobCensus.GetProximityCounter().getMobWithinThosesDistances(-1, 64 * 16);// MG TODO : deal with this magic number (the 16 is the size of a block)
-	for (cMobProximityCounter::tDistanceToMonster::const_iterator itr = allCloseEnoughToMoveMobs.m_Begin; itr != allCloseEnoughToMoveMobs.m_End; ++itr)
+	class cCallback : public cEntityCallback
 	{
-		itr->second.m_Monster.Tick(a_Dt, itr->second.m_Chunk);
-	}
+		virtual bool Item(cEntity * a_Entity) override
+		{
+			if (!a_Entity->IsMob())
+			{
+				return false;
+			}
+			cMonster * Monster = static_cast<cMonster*>(a_Entity);
+			cPlayer * Player = a_Entity->GetWorld()->FindClosestPlayer(a_Entity->GetPosition(), 128, false);
```

Comment:
Yes. There's lots of room for improvement, but I want animals to stop disappearing asap (#2991), and only optimize later.